### PR TITLE
feat: Add module to compress images based on quality #13

### PR DESCRIPTION
**Pull request by addityasingh**
- [x] Add module to compress images based on quality
- [x] Add docs

### CODE REVIEW

File: src/optimal_image/compress/image.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+impl<'a> Drop for VipsImage<'a> {
+    fn drop(&mut self) {
+        unsafe {
+            ffi::g_object_unref(self.c as *mut c_void);
+        }
+    }
+}
+
+impl<'a> VipsImage<'a> {
```
Comment (axyz) [inline src/optimal_image/compress/image.rs]: @addityasingh to keep it easy we can try it out reading and saving to file every time, but of course that would be pretty slow and we need to be able to seamlessly use in memory buffers. 

The first challenge would be to make sure we can read and write from raw buffers both as `VipsImage` and our `ImageData` format (`ImgVec<RGBA>`)

The second challenge would probably be to please the borrow checker doing that :dancer:

Comment (addityasingh) [reply src/optimal_image/compress/image.rs]: Agree. The borrow checker is a best in itself. But lets go ahead with the gradual approach and make it work. The in-memory buffers could be added as improvement once this works

**Comment by addityasingh** [comment]
@axyz Can you check if the JPEG image load (https://github.com/axyz/optimal-image/blob/a181e8896306c67af570c9253f6c3e2917f75d44/src/optimal_image/dataclients/jpeg.rs#L20-L35) performance works for you correctly ? for me it takes more than a minute to compare 2 jpegs of size 2.9MB and 1.5MB

**Comment by addityasingh** [comment]
I performed a basic comparison for images with original image of size 2.9MB and this is the result:

| Quality |size (in MB)| Size (% of original)| DSSIM score|
|--|--|--|--|
|100%| 2.9|100%| 0|
|90%|2.9|100%| 0.0021176602626094|
|85%|2.5|86.20%|0.004033526531560305|
|80%|2.2|75.86%|0.006118262269113828|
|75%|1.8|62.06%|0.008286237814875586|
|70%|1.5|51.72%|0.010396032637861374|
|50%|1.1|37.93%|0.016738859569819242|

**Comment by axyz** [comment]
@addityasingh the socres seems consistent. 

a 2.9mb image is pretty big... but did you try the comparison in production mode?

**Comment by addityasingh** [comment]
Seems to be working now. Will merge this one and create a new for the test and doc changes I have

**Pull request merged**